src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -611,26 +620,24 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-            // Load key-values registered for refresh that are not already loaded
-            await LoadKeyValuesRegisteredForRefresh(client, serverData, cancellationToken).ConfigureAwait(false);
             return serverData;
         }
 
-        private async Task LoadKeyValuesRegisteredForRefresh(ConfigurationClient client, IDictionary<string, ConfigurationSetting> data, CancellationToken cancellationToken)
+        private async Task<Dictionary<KeyValueIdentifier, ConfigurationSetting>> LoadKeyValuesRegisteredForRefresh(ConfigurationClient client, IDictionary<string, ConfigurationSetting> data, CancellationToken cancellationToken)

```

Comment:
I don't think the parameter name `data` is descriptive enough here. How about `existingSettings`?